driver-core/src/main/java/com/datastax/driver/core/querybuilder/Select.java

```diff
@@ -148,16 +153,15 @@ public Select limit(int limit) {
             throw new IllegalStateException("A LIMIT value has already been provided");
 
         this.limit = limit;
-        checkForBindMarkers(null);
```

Comment:
Looking at the history, it's a convoluted way to set `dirty = true` in the parent class. We need it for that scenario:
```java
Select s = select().all().from("foo");
System.out.println(s.getQueryString());
s = s.limit(10);
System.out.println(s.getQueryString());
```
It's also used in `Batch.add`. I suggest adding a `setDirty` method to make this more intuitive.